interfaces.go

```diff
@@ -93,10 +93,45 @@ func (b *Breadcrumb) MarshalJSON() ([]byte, error) {
 // User describes the user associated with an Event. If this is used, at least
 // an ID or an IP address should be provided.
 type User struct {
-	Email     string `json:"email,omitempty"`
-	ID        string `json:"id,omitempty"`
-	IPAddress string `json:"ip_address,omitempty"`
-	Username  string `json:"username,omitempty"`
+	ID        string            `json:"id,omitempty"`
+	Email     string            `json:"email,omitempty"`
+	IPAddress string            `json:"ip_address,omitempty"`
+	Username  string            `json:"username,omitempty"`
+	Name      string            `json:"name,omitempty"`
+	Segment   string            `json:"segment,omitempty"`
+	Data      map[string]string `json:"data,omitempty"`
+}
+
+func (u User) IsEmpty() bool {
+	if len(u.ID) > 0 {
+		return false
```

Comment:
I’m wonder if it’s more efficient to join the Boolean statements together performance wise then have all of the potential branches.

interesting to benchmark but I wouldn’t pursue this because it’s not worth the reduction in readability

Reply:
You mean `if ... || ...` ?

Reply:
Like `return … && … &&`

Reply:
Tried benchmarking this, no noticeable difference

```diff
diff --git a/interfaces.go b/interfaces.go
index e1b8b46..1e9fb17 100644
--- a/interfaces.go
+++ b/interfaces.go
@@ -134,6 +134,10 @@ func (u User) IsEmpty() bool {
 	return true
 }
 
+func (u User) IsEmptyWithBool() bool {
+	return len(u.ID) == 0 && len(u.Email) == 0 && len(u.IPAddress) == 0 && len(u.Username) == 0 && len(u.Name) == 0 && len(u.Segment) == 0 && len(u.Data) == 0
+}
+
 // Request contains information on a HTTP request related to the event.
 type Request struct {
 	URL         string            `json:"url,omitempty"`
diff --git a/interfaces_test.go b/interfaces_test.go
index 08b4d48..17f7793 100644
--- a/interfaces_test.go
+++ b/interfaces_test.go
@@ -40,6 +40,41 @@ func TestIsEmpty(t *testing.T) {
 	}
 }
 
+var tests = []struct {
+	name  string
+	input User
+}{
+	{name: "empty", input: User{}},
+	{name: "id", input: User{ID: "foo"}},
+	{name: "email", input: User{Email: "foo@example.com"}},
+	{name: "ip_address", input: User{IPAddress: "127.0.0.1"}},
+	{name: "username", input: User{Username: "My Username"}},
+	{name: "name", input: User{Name: "My Name"}},
+	{name: "segment", input: User{Segment: "My Segment"}},
+	{name: "datat", input: User{Data: map[string]string{"foo": "bar"}}},
+	{name: "all", input: User{ID: "foo", Email: "foo@example.com", IPAddress: "127.0.0.1", Username: "My Username", Name: "My Name", Segment: "My Segment", Data: map[string]string{"foo": "bar"}}},
+}
+
+func BenchmarkUserIsEmpty(b *testing.B) {
+	for _, v := range tests {
+		b.Run(v.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				v.input.IsEmpty()
+			}
+		})
+	}
+}
+
+func BenchmarkUserIsEmptyWithBool(b *testing.B) {
+	for _, v := range tests {
+		b.Run(v.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				v.input.IsEmptyWithBool()
+			}
+		})
+	}
+}
+
 func TestUserMarshalJson(t *testing.T) {
 	tests := []struct {
 		input User
```

which lead to:

```
BenchmarkUserIsEmpty/empty-10           256869885                4.656 ns/op
BenchmarkUserIsEmpty/id-10              257214072                4.685 ns/op
BenchmarkUserIsEmpty/email-10           257459233                4.747 ns/op
BenchmarkUserIsEmpty/ip_address-10      254609672                4.660 ns/op
BenchmarkUserIsEmpty/username-10        257343769                4.693 ns/op
BenchmarkUserIsEmpty/name-10            255480726                4.660 ns/op
BenchmarkUserIsEmpty/segment-10         257549162                4.656 ns/op
BenchmarkUserIsEmpty/datat-10           257294590                4.722 ns/op
BenchmarkUserIsEmpty/all-10             256479627                4.665 ns/op
BenchmarkUserIsEmptyWithBool/empty-10           257178012                4.667 ns/op
BenchmarkUserIsEmptyWithBool/id-10              257351770                4.666 ns/op
BenchmarkUserIsEmptyWithBool/email-10           257097589                4.672 ns/op
BenchmarkUserIsEmptyWithBool/ip_address-10      255344410                4.669 ns/op
BenchmarkUserIsEmptyWithBool/username-10        257379093                4.666 ns/op
BenchmarkUserIsEmptyWithBool/name-10            257207043                4.667 ns/op
BenchmarkUserIsEmptyWithBool/segment-10         254033016                4.688 ns/op
BenchmarkUserIsEmptyWithBool/datat-10           257323213                4.717 ns/op
BenchmarkUserIsEmptyWithBool/all-10             252587486                4.692 ns/op
```